### PR TITLE
docs: add Hugo home page, fix vulnerabilities, and expand test coverage

### PR DESCRIPTION
**Pull request by kcloutie**
- Add home/_index.md as the Hugo site landing page with install instructions,
  30-second example, core concepts table, 18 built-in providers, key features,
  documentation links, and a full CI/CD pipeline example
- Mount home/ directory in hugo.yaml and site/hugo.yaml so the landing page
  renders at the site root; add home/** to the pages.yml push trigger paths
- Add Documentation badge to README.md linking to the Hugo site

ci: consolidate and harden GitHub Actions workflows
- Merge pr-checks.yml into test.yml (renamed to CI); remove duplicate test runs
  on PRs by splitting into separate lint and test jobs
- Add concurrency groups to CI and CodeQL workflows to cancel superseded runs
- Add govulncheck step pinned to v1.1.4 to detect known CVEs in Go modules
- Add go mod tidy drift check and explicit go build step for fast failure signal
- Promote SCAFCTL_SECRET_KEY to job-level env in test job
- Add test gate (unit + integration) to release.yml before GoReleaser runs
- Add Syft install and sboms config to .goreleaser.yaml for SPDX SBOM generation
- Pin all action references to immutable commit SHAs across all workflows
- Delete redundant pr-checks.yml
- Pin GoReleaser CLI to v2.14.3 in release.yml
- Add goreleaser-action version input to pin binary version

fix: bump Go directive to 1.26.1 to resolve 5 stdlib CVEs
- GO-2026-4599 to GO-2026-4603 (crypto/x509, net/url, os, html/template)
- Updated go.mod, site/go.mod, and scripts/go.mod

fix(lint): correct testFileReachable path containment check for root solutionDir
- Replace HasPrefix(cleanedBase+sep) guard with filepath.Rel-based check so
  the function works correctly when solutionDir is the filesystem root (/)
- Add TestTestFileReachable_RootSolutionDir test to cover the edge case

refactor(auth/diagnose): inject http.Client and endpoint into RunClockSkewCheck
- Enables hermetic unit testing without real network calls
- Add httptest.Server-based tests covering all branches; gate real-network
  path behind -short flag

test: add comprehensive unit test coverage across 110+ packages
- New test files for action, auth, cache, catalog, celexp, config, gotmpl,
  httpc, lint, logger, paths, profiler, provider, resolver, schema, secrets,
  settings, shellexec, soldiff, solution, spec, telemetry, and terminal packages

### CODE REVIEW

File: .github/workflows/pages.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ on:
     branches: [main]
     paths:
       - "docs/**"
+      - "home/**"
       - "assets/**"
       - "layouts/**"
```
[the remark this answers is not in the document]
Comment (kcloutie) [reply .github/workflows/pages.yml]: Fixed — pinned all five actions in `pages.yml` to immutable commit SHAs: `actions/checkout` → v4.2.2, `actions/setup-go` → v5.5.0, `actions/configure-pages` → v5.0.0, `actions/upload-pages-artifact` → v3.0.1, `actions/deploy-pages` → v4.0.5.
